.github/workflows/run_all_frameworks.yml

```diff
@@ -0,0 +1,128 @@
+name: Run All Frameworks
+
+on: pull_request
+
+
+jobs:
+
+  detect_changes:  # Figure out which frameworks need to be evaluated based on which files are changed.
+    name: Detect Changes
+    runs-on: ubuntu-latest
+    outputs:
+      frameworks: ${{ steps.find-required-tests.outputs.frameworks }}
+      skip_baseline: ${{ steps.find-required-tests.outputs.skip_baseline }}
+      skip_evaluation: ${{ steps.find-required-tests.outputs.skip_evaluation }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: pull base branch
+        run: |
+          git fetch --all
+          git checkout -b $GITHUB_BASE_REF origin/$GITHUB_BASE_REF
+          git checkout $GITHUB_HEAD_REF
+      - id: find-required-tests
+        name: Detect Common Changes  # detect if any changes occurred that should trigger all frameworks.
+        run: |
+          changed_files=$(git diff --name-only HEAD..$GITHUB_BASE_REF)
+          common_files="amlb/** "\
+          "frameworks/shared/** "\
+          ".github/workflows/run_all_frameworks.yml "\
+          ".github/runbenchmark/action.yml "\
+          "runbenchmark.py "\
+          "requirements.txt"
```

Comment:
Not exactly sure what the best way to do this is. The idea is that we need to test all frameworks if shared code has changed, so we need a way to define shared code. I was also considering an alternative where everything is shared code unless it's specifically not, but that's not very clean either due to the `frameworks` folder which contains both shared and specific code. Note that we don't want to trigger this CI on e.g. doc changes.